front-end/app/service/basket.service.ts

```diff
@@ -33,13 +33,13 @@ export class BasketService extends SharedBasketService {
     removeAll(): void {
         super.removeAll();
         this.saveItems();
-        localStorage.removeItem(STORAGE_KEY);
     }
 
     private loadItems(): void {
         let basket = JSON.parse(localStorage.getItem(STORAGE_KEY) || null);
-        if (basket && basket.date) {
-            if (new Date().valueOf() - basket.date <= 21600000) {
+        const HOUR = 60 * 60 * 1000;
+        if (basket) {
```

Comment:
In our scenario, basket will either be an object or it will be null. I generally prefer to be as explicit as possible and write basket !== null because this avoids any coercion.